devops/nightly.yml

```diff
@@ -13,182 +19,275 @@ schedules:
+  dependsOn: []
+  jobs:
+  - job:
+    displayName: Linux
+    pool:
+      vmImage: 'ubuntu-20.04'
+    timeoutInMinutes: 120
+    strategy:
+      matrix:
+        GCC Release:
+          CC: gcc
+          CXX: g++
+          CXXFLAGS:
+          Asan: Off
+          BuildType: Release
+        Clang Debug (ASAN):
+          CC: clang
+          CXX: clang++
+          CXXFLAGS: -stdlib=libstdc++
+          BuildType: Debug
+          Asan: On
+
+    variables:
+      - template: ./vars/anybuild-vars.yml
+      - template: ./vars/llvm-vars.yml
+
+    steps:
+    - checkout: self
+
+    - script: |
+        set -eo pipefail
+        git submodule init
+        git submodule update --depth 1 --recursive
+      displayName: 'Checkout submodules'
+
+    - template: ./steps/linux/dependencies.yml
+
+    - bash: |
+        set -euo pipefail
+
+        # drop an AnyBuild.json config file and use it to
+        #   - ignore the CMAKE_BUILD_PARALLEL_LEVEL variable
+        #   - limit the number of concurrent cache lookups
+        echo '{"ActionCache": { "MaxParallelLookups": 10, "IgnoredEnvVars": [ "CMAKE_BUILD_PARALLEL_LEVEL" ] }}' | tee AnyBuild.json
+
+        # remove '-march=native' which is incompatible with remote execution
+        sed -i 's/ -march=native//g' src/rt/CMakeLists.txt
+
+        echo "=============="
+        echo "Patch applied:"
+        echo "=============="
+        echo
+        git diff
+      displayName: Configure AnyBuild
+
+    - bash: |
+        set -euo pipefail
+
+        mkdir -p build
+        cd build
+        cmake                                      \
+          ..                                       \
+          -GNinja                                  \
+          -DCMAKE_BUILD_TYPE=$(BuildType)          \
+          -DCMAKE_CXX_FLAGS=$(CXXFLAGS)            \
+          -DENABLE_ASSERTS=ON                      \
+          -DUSE_ASAN=$(Asan)                       \
+          -DVERONA_CI_BUILD=On                     \
+          -DRT_TESTS=ON                            \
+          -DVERONA_DOWNLOAD_LLVM=OFF               \
+          -DLLVM_EXTRA_CMAKE_ARGS="                \
+            -Wno-dev                               \
+            -DCMAKE_C_COMPILER=$(LLVM_CC)          \
+            -DCMAKE_CXX_COMPILER=$(LLVM_CXX)       \
+            -DCMAKE_BUILD_TYPE=$(LLVM_BuildType)   \
+            -DCMAKE_CXX_FLAGS=$(LLVM_CXXFLAGS)     \
+            -DLLVM_USE_SANITIZER=$(LLVM_Sanitizer) \
+            -DMLIR_INCLUDE_TESTS=OFF"
+      displayName: CMake
+
+    - template: ./steps/linux/anybuild.yml
+      parameters:
+        BuildName: Verona-$(BuildType)-$(CC)-Asan$(Asan)-AnyBuild
+        BuildCommandLine: env CMAKE_BUILD_PARALLEL_LEVEL=${{ parameters.AnyBuildParallelism }} cmake --build .
+        WorkingDirectory: build
+
+    - template: ./steps/linux/test.yml
+      parameters:
+        WorkingDirectory: build
 
 ############################################## Windows Builds
```

Comment:
everything below should be exactly the same, just indented to accommodate the newly introduced scopes